### ISSUE TITLE
[Object Page]: Accessibility issue: contentinfo landmark should not be contained in another landmark

### ISSUE THREAD

**Issue by bomasap** [SAP/ui5-webcomponents-react#7173]
### Describe the bug

While running Axe DevTools, we encountered the following accessibility issue:

> "Contentinfo landmark should not be contained in another landmark."

Root Cause:
In the current HTML structure:
```html
<div id="content">
  <footer>
    ...
  </footer>
</div>
``` 

The `<div id="content">` is explicitly assigned `role="main"` and `data-landmark-guessed`, which signals to accessibility tools that it serves as a main landmark.

As a result, the `<footer>` element (which defines a contentinfo landmark by default) becomes a descendant of the main landmark, which violates [WAI-ARIA landmark roles rules](https://www.w3.org/TR/wai-aria-practices/#aria_lh_contentinfo).

### Isolated Example

https://stackblitz.com/edit/github-t5j5mn-2urftkct?file=src%2FApp.tsx,package.json

### Reproduction steps

1. Access: https://stackblitz.com/edit/github-t5j5mn-2urftkct?file=src%2FApp.tsx,package.json
2. Run Axe DevTools(Ensure to keep the Best Practices toggle on while running the AXE tool.)
3. Observe Accessibility issue. 

### Screenshots or Videos

![Image](https://github.com/user-attachments/assets/b1a74aeb-6572-48c9-a22e-b5db3e19a0dd)
![Image](https://github.com/user-attachments/assets/431a307f-71ed-4c1f-a2ff-bb4a66dab248)

### UI5 Web Components for React Version

2.7.2

### UI5 Web Components Version

2.7.0

### Browser

Safari, Chrome, Edge, Firefox

### Relevant log output

```shell

```

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by bomasap** [comment]
Hi @Lukas742 

Thanks for your comment. 

You need to turn "Best Practices" Toggle on in Axe DevTools.

**Comment by Lukas742** [comment]
I did:

![Image](https://github.com/user-attachments/assets/b54b5ba2-5794-414f-8eff-f2d95467b1bd)

Also, please keep in mind that this is only an example. While it should be in general accessible, it doesn't mean it lives up to app standards. So, if an accessibility issue is found in one of our examples, we will gladly fix it, but for this the example needs to be isolated (meaning it should only use the components required to reproduce the issue). 
Also, axe showing an error doesn't mean it's an actual accessibility issue that is affecting users as it could be false positive, so please explain why users are affected by this. E.g. by providing the faulty screen reader announcement. 

>When reporting issues with different testing environments, please ensure that the issue is not false positive, a real accessibility concern is present, and there is an impact on the users.
_[contribution guidelines](https://sap.github.io/ui5-webcomponents/docs/advanced/accessibility/#testing-accessibility) for a11y_

**Comment by bomasap** [comment]
Hi Luaks,

Thank you for your reply. I’ve identified the root cause of the issue — the Chrome browser automatically adds `role="main"` to `div#content`, which leads to the axe error: “contentinfo landmark should not be contained in another landmark.” I’m currently discussing with the team whether we can remove the `#content` element altogether.

However, there's still another accessibility issue related to the UI5 Web Components. When the `FlexibleColumnLayout `component is assigned `role="main"`, and the `startColumn` contains an ObjectPage component, the same axe error appears: “contentinfo landmark should not be contained in another landmark.” The issue is caused by the `<footer>` element being nested inside another landmark. Could you please take a look?


If we need to remove role="main" from the FlexibleColumnLayout, could you advise where we should assign the role="main" instead?

I’ve also updated the isolation example — please take a look. https://stackblitz.com/edit/github-t5j5mn-2urftkct?file=src%2FApp.tsx,package.json

**Comment by Lukas742** [comment]
> the Chrome browser automatically adds role="main" to div#content, which leads to the axe error: “contentinfo landmark should not be contained in another landmark.”

I'm not aware that this is enabled by Chrome per default, since using heuristics for accessibility, such as inferring landmark roles based on an element’s id, is unreliable. Heuristics depend on assumptions that may vary between tools and browsers, leading to inconsistent behavior. Is this also happening when disabling all plugins you have installed and outside of AXE Dev tools?

> The issue is caused by the <footer> element being nested inside another landmark. Could you please take a look?

I drilled down the issue to it's core and you're right, here you can find a minimal reproducible example that shows the issue: https://stackblitz.com/edit/github-t5j5mn-pd4sm2rp?file=src%2FApp.tsx,package.json

I find it strange that Axe DevTools only reports an issue if `<div role="main">` is used instead of `<main>`, because they should give the same result, but as both should report the same error, we will offer a way to configure the `role` of the `footerArea` container. Once the linked PR is merged and a new version is published you can use the `accessibilityAttributes` prop like this for example: 
```
accessibilityAttributes={{ objectPageFooterArea: { role: 'presentation' } }}
```

**Comment by bomasap** [comment]
Hi @Lukas742 
Regarding the issue of div#content automatically adding role="main", I've identified that the 'Landmark Navigation' Chrome extension's functionality automatically adds role="main". Apologies for the earlier confusion.

![Image](https://github.com/user-attachments/assets/736816b1-f805-4e53-b32c-369da7bfb28b)

Thank you for providing the ability to resolve the <footer> element being nested inside another landmark issue; I will update it once the new version is published."
